app/models/duplicate_profile.rb

```diff
@@ -3,6 +3,8 @@
 class DuplicateProfile < ApplicationRecord
   def self.involving_profile(profile_id:, service_provider:)
     where(service_provider: service_provider)
-      .where('? = ANY(profile_ids)', profile_id).first
+      .where('? = ANY(profile_ids)', profile_id)
```

Comment:
We should be passing in an array of profile ids, and using `&&`. Imagine if a `duplicate_profiles` object exists with [1,2]. Now, if we discover that 3 is a duplicate of 1 and 2, we would not find the `duplicate_profiles` object if we only search with 3.